BEIMA.Backend/MongoService/Device.cs

```diff
@@ -96,17 +139,12 @@ public Device(ObjectId id, ObjectId deviceTypeId, string deviceTag, string manuf
         /// <param name="arg">The Property being null checked.</param>
         /// <param name="name">The name of the Property, to be returned in the exception message.</param>
         /// <exception cref="ArgumentNullException">Throws an exception when the passed in argument is null.</exception>
-        private void CheckNullArgument(dynamic arg, string name = "")
+        private void CheckNullArgument(Object arg, string name = "")
         {
             if (arg == null)
             {
                 throw new ArgumentNullException($"Device - {name} is null");
             }
-
-            if (arg is BsonDocument && arg.ElementCount == 0)
```

Comment:
Still need to check if SetLocation/SetLastModified have been called yet. Refer to Building.cs and User.cs files.

Reply:
It now checks to see if they were set